In [1]:
import numpy as np
import pandas as pd

let's simulate an impact factor for 2012

In [2]:
# Total counts of citations of those papers in 2012 / Npaperspublished is the IF
N = 1000
y2012_citations_of_papers_published_in_2010 = np.random.randint(0,30,N)
y2012_citations_of_papers_published_in_2011 = np.random.randint(0,30,N)

IF_2012 = (np.sum(y2012_citations_of_papers_published_in_2010) + np.sum(y2012_citations_of_papers_published_in_2011)) / (len(y2012_citations_of_papers_published_in_2010) + len(y2012_citations_of_papers_published_in_2011))
IF_2012

14.4135

In [3]:
# simplified formula has average and sum ~= 0
np.round(np.mean(np.concatenate((IF_2012 - y2012_citations_of_papers_published_in_2010,IF_2012 - y2012_citations_of_papers_published_in_2011))))

0.0

In [4]:
y2011_citations_of_papers_published_in_2010 = np.random.randint(0,30,N)
y2011_citations_of_papers_published_in_2009 = np.random.randint(0,30,N)

IF_2011 = (np.sum(y2011_citations_of_papers_published_in_2009) + np.sum(y2011_citations_of_papers_published_in_2010)) / (len(y2011_citations_of_papers_published_in_2009) + len(y2011_citations_of_papers_published_in_2010))
IF_2011

14.5185

_now i have the 2011 and 2012 IF factors of papers published in 2010, along with their citations:
 we're ready to calculate the BS_

In [8]:
# we need to invert numerator and denominator of bs formula (as suggested by Alex some time ago)
# BUT: an index < 1 is bad, while > 1 is good
print('**2010 bs indices**')
(c1+c2) / (IF_2011 + IF_2012)

**2010 bs indices**


array([0.96778653, 1.55537122, 1.86644546, 0.10369141, 1.00235034,
       0.44932946, 0.89865892, 0.79496751, 1.55537122, 1.07147795,
       1.31342458, 1.17516936, 1.34798839, 0.48389327, 0.7258399 ,
       1.34798839, 0.93322273, 1.31342458, 1.14060556, 1.4516798 ,
       1.86644546, 0.7258399 , 0.79496751, 0.44932946, 1.48624361,
       0.6912761 , 1.27886078, 0.34563805, 0.65671229, 0.93322273,
       1.07147795, 0.24194663, 1.03691414, 0.86409512, 1.14060556,
       1.52080741, 1.20973317, 0.44932946, 0.82953131, 0.20738283,
       1.52080741, 1.03691414, 0.89865892, 0.86409512, 1.20973317,
       1.17516936, 0.6912761 , 0.34563805, 0.58758468, 1.55537122,
       1.97013687, 1.4516798 , 1.27886078, 0.7258399 , 1.00235034,
       1.14060556, 0.86409512, 0.7258399 , 0.34563805, 0.51845707,
       1.14060556, 1.62449882, 1.07147795, 1.38255219, 1.00235034,
       1.83188165, 0.55302088, 1.69362643, 0.58758468, 1.27886078,
       0.93322273, 1.17516936, 1.34798839, 0.82953131, 1.10604

In [9]:
# then average BS of the publications is indeed ~= 1 
# negligible distance -within a decimal point- arise with N of publications is low
print('average bs')
np.round(np.mean((c1+c2) / (IF_2011 + IF_2012)),2)

average bs


1.0

In [10]:
# also if we have a publication with 0 citations (first value in array) in both years, 
# that's fine because it stays 0, ergo smallest number in the bunch!
c1[0] = 0
c2[0] = 0

In [13]:
(c1+c2) / (IF_2011 + IF_2012)

array([0.        , 1.55537122, 1.86644546, 0.10369141, 1.00235034,
       0.44932946, 0.89865892, 0.79496751, 1.55537122, 1.07147795,
       1.31342458, 1.17516936, 1.34798839, 0.48389327, 0.7258399 ,
       1.34798839, 0.93322273, 1.31342458, 1.14060556, 1.4516798 ,
       1.86644546, 0.7258399 , 0.79496751, 0.44932946, 1.48624361,
       0.6912761 , 1.27886078, 0.34563805, 0.65671229, 0.93322273,
       1.07147795, 0.24194663, 1.03691414, 0.86409512, 1.14060556,
       1.52080741, 1.20973317, 0.44932946, 0.82953131, 0.20738283,
       1.52080741, 1.03691414, 0.89865892, 0.86409512, 1.20973317,
       1.17516936, 0.6912761 , 0.34563805, 0.58758468, 1.55537122,
       1.97013687, 1.4516798 , 1.27886078, 0.7258399 , 1.00235034,
       1.14060556, 0.86409512, 0.7258399 , 0.34563805, 0.51845707,
       1.14060556, 1.62449882, 1.07147795, 1.38255219, 1.00235034,
       1.83188165, 0.55302088, 1.69362643, 0.58758468, 1.27886078,
       0.93322273, 1.17516936, 1.34798839, 0.82953131, 1.10604

**Yook and Zook simulations:**

In [44]:
# identify the high and low bounds of the citation distributions
tot_cit = c1+c2
lower_bound = np.percentile(tot_cit, 10)
upper_bound = np.percentile(tot_cit, 90)

In [53]:
print('average Yook BS:',(np.round(np.mean(tot_cit[(c1+c2)<=lower_bound])/ (IF_2011 + IF_2012),2)))

average Yook BS: 0.31


In [54]:
print('average Zook BS:',(np.round(np.mean(tot_cit[(c1+c2)>=upper_bound])/ (IF_2011 + IF_2012),2)))

average Zook BS: 1.57


**i find that the range is kind of restricted now: papers that have up to 4 times the expected citations
 get a bs index of 1.97 at most, while lowest is 0 which is the same as citations so a bit redundant**